### PR TITLE
feat: run unit tests against the bigtable emulator

### PR DESCRIPTION
**Pull request by pjenvey**
also make BigTableError more verbose and use the try! operator in the run_gauntlet test

Closes: SYNC-4059

### CODE REVIEW

**Review by jrconlin — APPROVED** [review]

File: tests/integration/test_integration_all_rust.py [hunk 1]
```diff
@@ -120,7 +121,7 @@ def get_db_settings() -> str | dict[str, str | int | float] | None:
             router_table=ROUTER_TABLE,
             message_table=MESSAGE_TABLE,
             current_message_month=MESSAGE_TABLE,
-            table_name="projects/test/instances/test/tables/autopush",
+            table_name="projects/test/instances/localhost:8086/tables/autopush",
```
[the remark this answers is not in the document]
Comment (pjenvey) [reply tests/integration/test_integration_all_rust.py]: Switched everything back to project "test" which apparently now works for me, oops.

File: scripts/setup_bt.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh -eu
```
[the remark this answers is not in the document]
Comment (pjenvey) [reply scripts/setup_bt.sh:1]: 👍 I had already made similar defaults for the other fields for a separate PR, brought that all here

File: autopush-common/src/db/bigtable/bigtable_client/error.rs [hunk 3]
```diff
@@ -75,41 +77,40 @@ impl ToString for MutateRowStatus {
...
-        .to_owned()
+        })
     }
 }
 
 #[derive(Debug, Error)]
 pub enum BigTableError {
-    #[error("Invalid Row Response")]
-    InvalidRowResponse(grpcio::Error),
+    #[error("Invalid Row Response: {0}")]
```
Comment (jrconlin) [inline autopush-common/src/db/bigtable/bigtable_client/error.rs:86]: Won't these produce errors that are harder to collect in Sentry? 
The argument is being displayed in the format and as part of the event, so it's not getting lost.

Comment (pjenvey) [reply autopush-common/src/db/bigtable/bigtable_client/error.rs:86]: I don't think this really complicates Sentry's collection -- we do [the same in syncstorage](https://github.com/mozilla-services/syncstorage-rs/blob/cc8d5847dc37de321a484321b222e0757bc322ab/syncstorage-spanner/src/error.rs#L66)

Outside of Sentry this is helpful, e.g. when diagnosing a db error locally which you typically aren't using Sentry with.

Comment (jrconlin) [reply autopush-common/src/db/bigtable/bigtable_client/error.rs:86]: Fair.

**Review by jrconlin — APPROVED** [review]

**Pull request merged**